src/features/staking/TransactionTracker.tsx

```diff
@@ -0,0 +1,87 @@
+import { ReactNode, useEffect } from "react";
+import { twJoin } from "tailwind-merge";
+import { Hash } from "viem";
+import { useWaitForTransaction } from "wagmi";
+import loadingSpinner from "../../assets/loading-spinner.svg";
+import { Button } from "../common/Button";
+import { EtherscanUrl } from "../common/EtherscanUrl";
+import { useStakingModalStore } from "./store/useStakingModalStore";
+import { ActionButton, Status } from "./types/StakingTypes";
+import { transactionTrackerIcon } from "./utils/iconTransactionTracker";
+
+export const TransactionTracker = ({
+  actionDescription,
+  successText,
+  actionButtons,
+  txHash,
+}: {
+  actionDescription?: string;
+  successText: string | ReactNode;
+  actionButtons?: ActionButton;
+  txHash?: Hash;
+}) => {
+  const { setTxHash } = useStakingModalStore();
+  const { data, status, isError } = useWaitForTransaction({
+    hash: txHash,
+    enabled: !!txHash,
+  });
+
+  const icon = transactionTrackerIcon(status);
+  const etherscanUrl = EtherscanUrl(txHash);
+
+  const shouldButtonRender = status === "success" || isError;
+
+  const buttonContents = (status: Status) => {
+    switch (status) {
+      case "success":
+        return {
+          text: actionButtons?.afterSuccess.label,
+          action: actionButtons?.afterSuccess.callback,
+        };
+      case "error":
+        return {
+          text: actionButtons?.afterFailure.label,
+          action: actionButtons?.afterFailure.callback,
+        };
+      default:
+        return;
+    }
+  };
+  const buttonContent = buttonContents(status);
+
+  useEffect(() => {
+    txHash ? setTxHash(data?.blockHash) : setTxHash(undefined);
+  }, [txHash, setTxHash, data?.blockHash]);
+
+  return (
+    <div className="flex flex-col items-center text-gray-500 pt-4">
+      <div
+        className={twJoin(
+          icon === loadingSpinner ? "m-auto animate-spin p-0" : "p-2",
+          "bg-black rounded-full",
+        )}
+      >
+        {icon && <img src={icon} alt="" />}
+      </div>
+      {successText && <div className="-mb-6">{successText}</div>}
```

Comment:
This variable is called `successText`, but is shown whether successful or not.

This is because you're currently implementing the logic to determine whether or not to show the text in `AmountStakedText`, which means you also have to track the status of the transaction in `StakingModal`. I think it would be cleaner to have the `TransactionTracker` track the status of the transaction, then conditionally display the success text only when successful. So you can then remove `txStatus` from `AmountStakedText`,